### ISSUE TITLE
`newrelic_notification_channel`: consecutive applies throw error messages

### ISSUE THREAD

**Issue by louhnmn** [newrelic/terraform-provider-newrelic#2009]
> :warning: **Important:** Failure to include the following, such as omitting the Terraform configuration in question, may delay resolving the issue.

- [x] Your New Relic `provider` [configuration](#terraform-configuration-files) (sensitive details redacted)
- [x] A list of [affected resources](#affected-resources) and/or data sources
- [x] The [configuration](#terraform-configuration-files) of the resources and/or data sources related to the bug report (i.e. from the list mentioned above)
- [x] Description of the [current behavior](#actual-behavior) (the bug)
- [x] Description of the [expected behavior](#expected-behavior)
- [x] Any related [log output](#debug-output)

### Terraform Version

```
terraform -v
Terraform v1.2.9
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v4.30.0
+ provider registry.terraform.io/newrelic/newrelic v3.2.1
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `newrelic_notification_channel`
- `newrelic_workflow`

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration
> Please include your `provider` configuration (sensitive details redacted) as well as the configuration of the resources and/or data sources related to the bug report.

`provider.tf`:

```hcl

terraform {
  required_version = "~> 1.2.9"

  required_providers {
    newrelic = {
      source  = "newrelic/newrelic"
      version = "~> 3.2.1"
    }
  }
}

provider "newrelic" {
  region = var.newrelic_region
}

```

`alerts.tf`:

```hcl


resource "newrelic_notification_destination" "google_chat_temp" {
  name = "TF Google Chat"
  type = "WEBHOOK"

  property {
    key   = "url"
    value = "https://chat.googleapis.com/v1/*redacted*"
  }
}

resource "newrelic_notification_channel" "google_chat_channel" {
  name           = "TF Google Chat Channel"
  type           = "WEBHOOK"
  destination_id = newrelic_notification_destination.google_chat_temp.id
  product        = "IINT"

  property {
    key = "payload"
    value = "{\"text\": \"test\"}"
    label = "Payload Template"
  }

  depends_on = [
    newrelic_notification_destination.google_chat_temp
  ]
}

resource "newrelic_workflow" "ssl_alert_workflow" {
  name                  = "SSL Cert Workflow"
  enabled               = true
  destinations_enabled  = true
  muting_rules_handling = "DONT_NOTIFY_FULLY_MUTED_ISSUES"

  destination { channel_id = newrelic_notification_channel.google_chat_channel.id }

  issues_filter {
    name = "filter-name"
    type = "FILTER"

    predicate {
      attribute = "labels.policyIds"
      operator  = "EXACTLY_MATCHES"
      values    = ["${newrelic_alert_policy.ssl_certificates.id}"]
    }
  }


  depends_on = [
    newrelic_notification_channel.google_chat_channel
  ]
}

resource "newrelic_alert_policy" "ssl_certificates" {
  name                = "SSL Certificates PROD"
  incident_preference = "PER_CONDITION_AND_TARGET"
}

```

### Actual Behavior
What actually happened?

- With the first apply channel and workflow get created.
- With the second apply, the workflow gets destroyed and recreated, throwing an `Error: INVALID_PARAMETER: A channel with id was not found`
- With the third apply it throws another error saying `[ERROR] notification channelResponse.Entities response is empty`

### Expected Behavior
What should have happened?

- the workflow should not get recreated on the second apply
- the workflow should find the channel_id after the third apply

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Debug Output

```
  # newrelic_workflow.ssl_alert_workflow will be created
  + resource "newrelic_workflow" "ssl_alert_workflow" {
      + destinations_enabled  = true
      + enabled               = true
      + id                    = (known after apply)
      + last_run              = (known after apply)
      + muting_rules_handling = "DONT_NOTIFY_FULLY_MUTED_ISSUES"
      + name                  = "SSL Cert Workflow"
      + workflow_id           = (known after apply)

      + destination {
          + channel_id = "*redacted_channel_id_will_be_used_later*"
          + name       = (known after apply)
          + type       = (known after apply)
        }

      + issues_filter {
          + filter_id = (known after apply)
          + name      = "filter-name"
          + type      = "FILTER"

          + predicate {
              + attribute = "labels.policyIds"
              + operator  = "EXACTLY_MATCHES"
              + values    = [
                  + "*redacted*",
                ]
            }
        }
    }

Plan: 1 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

newrelic_notification_destination.google_chat_temp: Modifying... [id=*redacted*]
newrelic_workflow.ssl_alert_workflow: Creating...
newrelic_notification_destination.google_chat_temp: Modifications complete after 1s [id=*redacted*]
newrelic_workflow.ssl_alert_workflow: Creation complete after 2s [id=*redacted*]

Apply complete! Resources: 1 added, 1 changed, 0 destroyed.
```

- consecutive `terraform apply`
- terraform tries to destroy the workflow
- then recreate it

```
╷
│ Error: INVALID_PARAMETER: A channel with id *redacted_channel_id_from_before* was not found
│ 
│   with newrelic_workflow.ssl_alert_workflow,
│   on alerts.tf line 35, in resource "newrelic_workflow" "ssl_alert_workflow":
│   35: resource "newrelic_workflow" "ssl_alert_workflow" {
│ 
╵
```

consecutive `terraform apply`

```
╷
│ Error: [ERROR] notification channelResponse.Entities response is empty
│ 
│ 
╵
```

- further applies will trigger this error afterwards every time
- to make it disappear `terraform state rm newrelic_notification_channel.*the_channel_name*`

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- #1974 
- #2002

**Comment by NSSPKrishna** [comment]
Hi @louhnmn thanks for raising the issue. We will prioritise it.

**Comment by pgalvarez-elsevier** [comment]
FWIW this still happens on 3.3.0

**Comment by IliaShurygin** [comment]
Hi! Thanks for the report!

This is the same issue as reported in https://github.com/newrelic/terraform-provider-newrelic/issues/2002
We have confirmed the problem and will work on a fix. 
As a temporary workaround, please explicitly add account id to the workflow resource. This is not a perfect solution, but it should at least allow you to create and manage the workflows until the issue is fixed.

**Comment by ayk33** [comment]
I don't know if this is related but I am also trying to create a "newrelic_notification_channel" resource and getting a very vague error 
```
╷
│ Error: UNKNOWN_ERROR: An unknown error has occurred
│
│   with newrelic_notification_channel.channels["xxxxx"],
│   on notification_channel.tf line 1, in resource "newrelic_notification_channel" "channels":
│    1: resource "newrelic_notification_channel" "channels" {
│
╵
ERRO[0013] 1 error occurred:
	* exit status 1
```

**Comment by IliaShurygin** [comment]
@ayk33, if possible, could you please file a separate GH issue for this? 
It's hard to say for sure without additional investigation, but it looks like a separate problem to me. 

If you do file a separate GH issue, I would greatly appreciate if you could include a terraform config that reproduces the problem.

Improving vague error messages is also something we plan to tackle later on, but we'd like to address other major issues first

**Comment by ayk33** [comment]
@IliaShurygin I created a new issue https://github.com/newrelic/terraform-provider-newrelic/issues/2053.

**Comment by IliaShurygin** [comment]
The problem with account_id is fixed in v3.5.0, you should now be able to create (was already possible before) and update (the previously broken part) workflows without specifying account_id in workflow resource explicitly.

Please note that the fix for the other errors you are seeing after trying to fix the state ("notification channelResponse.Entities response is empty") is still in review and is not released yet.
